File: solutions/cpp/resistor-color/1/resistor_color.cpp
#include "resistor_color.h"
#include<stdexcept>

using namespace std;

namespace resistor_color {

     vector<string> const _colors{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};

    int color_code(string color){
        for(auto i=0;i<10; i++){
            if(color == _colors.at(i))
                return i;
        }
        throw std::invalid_argument("Invalid color.");
    }
    
    vector<string> colors() {return _colors;};
    

}  // namespace resistor_color

File: solutions/julia/resistor-color/1/resistor-color.jl
color_dict = 
    Dict("black"=>0, "brown"=>1, "red"=>2, "orange"=>3,
        "yellow"=>4, "green"=>5, "blue"=>6, "violet"=>7,
        "grey"=>8, "white"=>9)

function colorcode(color)
    color_dict[color]
end

function colors()
    sort(collect(keys(color_dict)); lt=(a,b) -> color_dict[a] < color_dict[b]) 
 end